realm/src/main/java/io/realm/internal/RealmCore.java

```diff
@@ -94,8 +93,10 @@ public static void print(String caption, AbstractCursor<?> cursor) {
     }
 */
 
-    public static void loadLibrary() {
-        if (libraryIsLoaded.get())
+    // Although loadLibrary is synchronized internally from AOSP 4.3, for the compatibility reason,
+    // KEEP synchronized here for the old devices!
+    public static synchronized void loadLibrary() {
+        if (libraryIsLoaded)
             // only load library once
```

Comment:
actually the comment here may be misleading, the lib will be loaded once anyway, this check will only save us one more trip to native land.